packages/simple-validators/src/validators.ts

```diff
@@ -207,22 +206,122 @@ export function validateEnum<T>(
+const ISO8601_DATE_PATTERN = /^(\d\d\d\d)(?:-(\d\d))?(?:-(\d\d))?(T.*)?$/;
+// This pattern implements the following ABNF from RFC2822: `[ day-of-week "," ] date`
+// with a trailing capture group for time-like information
+const RFC2822_DATE_PATTERN =
+  /^(?:(Mon|Tue|Wed|Thu|Fri|Sat|Sun),)?\s*(\d{1,2})\s+(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s+(\d\d\d\d)\s*([^\s].*)?$/;
+
+const MONTH_TO_NUMBER = new Map(
+  ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'].map(
+    (elem, index) => [elem, index + 1],
+  ),
+);
+
+function buildISO8601DateString(year: number, month: number, day: number): string {
+  const paddedMonth = `${month}`.padStart(2, '0');
+  const paddedDay = `${day}`.padStart(2, '0');
+  return `${year}-${paddedMonth}-${paddedDay}`;
+}
+
 /**
  * Validate date string or object
  *
- * Uses javascript Date() constructor; any input to the constructor that returns
- * a valid date is valid input. This includes ISO 8601 formatted strings and
- * IETF timestamps are valid.
+ * Parses strings as ISO 8601 dates, or a variant of RFC 2822 dates, falling back to the Date constructor otherwise.
+ * Parses Date objects as UTC or local dates according to the given options.
+ * Parses `null` dates as "now".
  */
-export function validateDate(input: any, opts: ValidationOptions) {
-  const date = new Date(input);
-  if (!date.getDate()) {
+export function validateDate(input: any, opts: ValidationOptions & { dateIsUTC?: boolean }) {
+  // String format dates
+  if (typeof input === 'string') {
+    // Try ISO 8601
+    let match = input.match(ISO8601_DATE_PATTERN);
+    if (match) {
+      const [year, month, day, tail] = match.slice(1, 5);
+      // Is a timestamp component present?
+      if (tail !== undefined) {
+        validationWarning(
+          `invalid date "${input}" - ISO 8601 dates must not include a time component ("${tail}"), which has been ignored`,
+          opts,
+        );
+      }
+      const fullMonth = parseInt(month);
+      const fullDay = parseInt(day);
+      if (fullMonth < 1 || 12 < fullMonth || fullDay < 1 || 31 < fullDay) {
+        return validationError(
+          `invalid date "${input}" - must be ISO 8601 or (specialised) RFC 2822 format calendar date`,
+          opts,
+        );
+      }
+      return [year, month, day].filter((item) => item).join('-');
+    }
+
+    // Try a variant of RFC2822
+    match = input.match(RFC2822_DATE_PATTERN);
+    if (match) {
+      const [day, month, year, tail] = match.slice(2, 6);
+
+      // Is a timestamp component present?
+      if (tail !== undefined) {
+        validationWarning(
+          `invalid date "${input}" - specialised RFC 2822 dates must not include a time component ("${tail}"), which has been ignored`,
+          opts,
+        );
+      }
+
+      const numericYear = parseInt(year);
+
+      // Convert Jan to 1 etc.
+      const numericMonth = MONTH_TO_NUMBER.get(month)!;
```

Comment:
Is there a reason we didn't just do `new Date(str)` if it didn't pass the initial checks?

I think that if people are doing their own thing, and not following the standard `YYYY-MM-DD` then we should just use the default javascript parser rather than writing our own?

Regardless, if we are adding our own simple date parsing, I would love to see some test coverage of these.

Reply:
Yes -- the RFC 2822 handling is not "standard" -- we're building a new rule from the ABNF grammar for RFC2822. That is, it's defined by the grammar in that RFC but we're not using their `date-time` rule. So, we need to roll our own.

We also want to identify rogue time-like components.